docs/CODE_GUIDELINES.md

```diff
@@ -764,4 +860,44 @@ Try to avoid using C macros. In many cases, they can be easily substituted with
 
 Prefer `constexpr` over `const` for constants when possible. Try to mark functions `constexpr` when reasonable.
 
+### 12.11. `std::string` vs `std::string_view`
+
+Prefer `std::string_view` over `std::string` when reasonable. Good examples are constants or method arguments. In the last case, it is not required declare the argument as reference or const, since the data source of string views are immutable by definition. A bad example is when you need a NUL-terminated C string, e.g. to interact with a C API. `std::string_view` does not offer a `c_str()` function like `std::string` does, but you can use `data()` to get the raw source of the data.
+
+Main reasons why we prefer `std::string_view` are: execution performance, no memory allocations, substrings can be made without copy, and the possibility to reuse the same data without reallocation.
+
+Avoid using `std::string_view` when you are not sure where the source of data is allocated, or as return value of a method. If not handled properly, the source (storage) of the data may go out of scope. As a consequence, the program enters undefined behavior and may crash, behave strangely, or exhibit security-relevant issues.
+
+✅ Good:
+```cpp
+namespace
+{
+constexpr std::string_view CONSTANT_FOO{"foo-bar"};
+} // unnamed namespace
+
+void CClass::SetText(std::string_view value) // using std::string_view you always use the same data source without new allocations 
+{
+  CLog::LogF(LOGDEBUG, "My name is {}", value);
+}
+
+[...]
+SetText(CONSTANT_FOO.substr(0, 3)); // no allocations made by substr, you return only a modified view of current string_view, with the same data source 
+```
+
+❌ Bad:
+```cpp
+namespace
+{
+constexpr std::string CONSTANT_FOO{"foo-bar"}; // no std::string_view use, we can avoid new allocations see `substr` used below 
+} // unnamed namespace
+
+void CClass::SetText(const std::string& value) // despite its declared as reference, std::string perform always some memory allocation, because there is no use of raw data
```

Comment:
I'm not sure I get it. How would, in the case written down here, std::string *always* cause a memory allocation, even if you passed in an existing std::string object by reference?

Reply:
its overhead, i will try reword it
from the net there are some explanations like this, is a bit technical to explain:
"""
use std::string_view over std::string const& as a function argument is that if you pass a string literal,
the latter will construct a heap-allocated temporary std::string object just for the function to operate on, and then free it again, whereas with std::string_view it's just as lightweight as passing by const char* (effectively free). If you already have your string data in a std::string object before passing it makes little difference apart from an extra indirection, but using std::string_view allows your function to be called with low overhead using either a std::string or a string literal
"""

Reply:
Yeah that is closer to my understanding :) Sorry to be pedantic, but I think that the guidelines should not make statements that might be misleading (such as the "always performs some memory allocation"). If you just shorten it to something along the lines that it will avoid memory allocations in many contexts such as passing C-string literals, that would already settle it for me.

Reply:
i.e., people can google the full explanation if they want it (or you link to it), but what we *do* write in the guidelines should be factually correct

Reply:
i changed them all, try check